docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,233 @@
+- The `Azure.*` package you'd like to use.
+
+The sample code in this article will use Key Vault secrets and Blob Storage for demonstration purposes.
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.Azure
+dotnet add package Azure.Identity
+dotnet add package Azure.Security.KeyVault.Secrets
+dotnet add package Azure.Storage.Blobs
+```
+
+## Register client
+
+In the `ConfigureServices` method, register a client for each service:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    // Add a KeyVault client
+    builder.AddSecretClient(keyVaultUrl);
+
+    // Add a Storage account client
+    builder.AddBlobServiceClient(storageUrl);
+
+    // Use DefaultAzureCredential by default
+    builder.UseCredential(new DefaultAzureCredential());
+  });
+
+  services.AddControllers();
+}
+```
+
+In code above, you need to explicitly specify the `keyVaultUrl` and `storageUrl`. Both variables are `Uri` types. The [Store configuration separately from code](#store-configuration-separately-from-code) section shows how you can avoid specifying the urls explicitly.
+
+The code above uses `DefaultAzureCredential` for authentication. [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential) will choose the best authentication mechanism based on your environment, allowing you to move your app seamlessly from development to production with no code changes.
+
+## Use the registered clients
+
+With the clients registered in `Startup`, you can now use them:
+
+```csharp
+[ApiController]
+[Route("[controller]")]
+public class MyApiController : ControllerBase
+{
+  private readonly BlobServiceClient blobServiceClient;
+
+  public MyApiController(BlobServiceClient blobServiceClient)
+  {
+    this.blobServiceClient = blobServiceClient;
+  }
+
+  /// Get a list of all the blobs in the demo container
+  [HttpGet]
+  public async Task<IEnumerable<string>> Get()
+  {
+    var containerClient = this.blobServiceClient.GetBlobContainerClient("demo");
+    var results = new List<string>();
+    await foreach (BlobItem blob in containerClient.GetBlobsAsync()) 
+    {
+      results.Add(blob.Name);
+    }
+    return results.ToArray();
+  }
+}
+```
+
+## Store configuration separately from code
+
+Recall in the [Register client](#register-client) section above, you explicitly specify the `keyVaultUrl` and `storageUrl`. This could cause problems when you run code against different environments during development and production. The ASP.NET Core team suggests [storing such configurations in environment dependent JSON files](/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#default-configuration). Thus, you can have an `appsettings.Development.json` file with one set of settings and an `appsettings.Production.json` with another set of configurations. The format of the file is:
```

Comment:
```suggestion
In the [Register client](#register-client) section, you explicitly specify the `keyVaultUrl` and `storageUrl`. This could cause problems when you run code against different environments during development and production. The .NET team suggests [storing such configurations in environment dependent JSON files](/dotnet/core/extensions/configuration-providers#json-configuration-provider). Thus, you can have an _appsettings.Development.json_ file with one set of settings and an _appsettings.Production.json_ with another set of configurations. The format of the file is:
```

We should be pointing more broadly to the .NET config docs here. The Azure .NET SDK is not only used in ASP.NET Core apps, it's applicable to all .NET workloads and this could be misleading otherwise.